fiatconnect-api.md

```diff
@@ -391,105 +408,238 @@ The FiatConnect API specifies *two* types of authentication; the first authentic
+
+Once the client generates a SIWE-compliant plaintext message, it signs it with the user's externally-owned account private key according to [EIP-191](https://eips.ethereum.org/EIPS/eip-191)
+`personal_sign` format. The client will then send the message and signature to the server at a `POST /auth/login` endpoint, described below. If the server
+accepts the signed message as valid, it will respond with a session cookie linked that identifies the user's authenticated session within the server. The server
+MUST set the [SameSite](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite) policy on the session cookie to prevent the client
+from sending the cookie to other hosts. Ultimately, it is the client's responsibility to manage session cookies between different FiatConnect APIs.
+
+If a client sends a session cookie when calling the `POST /auth/login` endpoint (either to refresh their current, valid session, or to generate a new session if their current one is
+expired), the server MUST ignore the provided cookie. If the authentication request is valid, the server MUST set a new value on the cookie referencing the client's new, valid session,
+and return it to the client.
+
+#### 3.2.1.5. `POST /auth/login`
+
+The `POST /auth/login` endpoint is responsible for verifying signed messages sent by clients, creating and authenticating a user's session, and returning session cookies.
+Note that the Sign-In With Ethereum standard, and FiatConnect, support authorization for externally owned accounts (EOAs) and contract-owned accounts; The `POST /auth/login`
+endpoint MUST honor login requests for EOAs, and MAY support smart contract-owned accounts as well, as described below.
+
+##### 3.2.1.5.1. Parameters
+
+###### 3.2.1.5.1.1. Request Body
+
+The request body must contain the following fields:
+
+* `message`: {`string`} [REQUIRED]
+  - The plaintext SIWE message
+* `signature`: {`object`} [REQUIRED]
+  - The [EIP-191](https://eips.ethereum.org/EIPS/eip-191) signed message if logging in as EOA, or a signature according to the
+	[EIP-1271](https://eips.ethereum.org/EIPS/eip-1271) standard, if logging in as a contract-owned account.
+
+The `message` field should be the *plaintext* message, while the signature should be an object containing the typical secp256k1 signature fields: `r`, `s`, and `v`.
+`r` and `s` are hex strings (prefixed with `0x`) each 32 bytes in length, and `v` is an integer. This is the "canonical" form of the signature as referenced in
+[EIP-2098](https://eips.ethereum.org/EIPS/eip-2098), rather than the "compact" form.
+
+##### 3.2.1.5.2. Responses
+
+###### 3.2.1.5.2.1. HTTP `200`
+
+On success, the server MUST return an HTTP `200`, update the state of the user's session to be authenticated, and return a session cookie.
+
+###### 3.2.1.5.2.2. HTTP `401`
+
+On failure, the server MUST return an HTTP `401`, with the following response body.
+
+```
 {
-	"alg": "ES256",
-	"typ": "JWT"
+	error: `ErrorEnum`,
```

Comment:
Would this just include a single enum, or do we also want to have a more descriptive error? E.g., which parameter is missing in case of `InvalidParameters`

Reply:
oooo the UI would love to know which parameter is at fault ❤️

Reply:
None of these SIWE parameters are actually chosen by the user; this is all done under the hood -- what would we display to the user if e.g., the nonce was wrong? Although for the client's sake, it might make sense to have more descriptive error types here.

Reply:
Just added a couple of extra error types... if there's more that would be useful to have, let me know and I can add them in!